### PR TITLE
Modify validates_overlap to accept IDs in the form of UUIDs as well as regular IDs 

### PR DESCRIPTION
**Pull request by rmathew1992**
In our application certain primary_keys are stored in the form of UUIDs. Unlike regular IDs UUIDs are strings.

Unfortunately this is not yet supported while updating a record. Currently "....!= #{record.send(primary_key_name)}" inserts the primary key value stripping it of quotation marks.

This PR checks to see if the primary key should be a string. If so it inserts them.

### CODE REVIEW

**Comment by robinbortlik** [comment]
Hello @rmathew1992 , first of all thanks a lot for your hard work. I your PR is really usefull. I will try to look on it as soon as possible and if everything will be ok, I will be very happy to accept it. 

Ps: can you please check, that the specs are passing locally for you, because it seems that they are failing on TravisCI.

File: lib/validates_overlap/overlap_validator.rb [hunk 1]
```diff
@@ -90,19 +90,35 @@ def primary_key(record)
...
     primary_key_name = primary_key(record)
+    key = primary_key_value(primary_key_name, record)
     if record.new_record?
       self.sql_conditions = main_condition
     else
-      self.sql_conditions = "#{main_condition} AND #{record_table_name(record)}.#{primary_key_name} != #{record.send(primary_key_name)}"
+      if primary_key_type(record) == :string
+        self.sql_conditions = "#{main_condition} AND #{record_table_name(record)}.#{primary_key(record)} != '#{key}'"
+      else
+        self.sql_conditions = "#{main_condition} AND #{record_table_name(record)}.#{primary_key(record)} != #{key}"
```
Comment (robinbortlik) [inline lib/validates_overlap/overlap_validator.rb]: Can you please refactor lines 115 and 117 to something like

```self.sql_conditions = "#{main_condition} AND #{record_table_name(record)}.#{primary_key(record)} !=```
```self.sql_conditions +=  primary_key_type(record) == :string ? '#{key}' : #{key}```

so we will remove the duplication.

**Comment by rmathew1992** [comment]
@robinbortlik  Thanks! Sorry it took so long for me to update, it took me a while to get all the tests to pass on Travis. *phew*

Let me know if there are any other changes you want me to make. :)

**Comment by robinbortlik** [comment]
Hi @rmathew1992 , I have to say, you did a very well job. :+1:  I have only one small, last requirement. It seems, that method 
```ruby
def primary_key_type(primary_key_name, record)
    record.class.for_attribute(primary_key_name)
end
``` 
is never used. Can you please check it and remove it if it is true?

**Comment by rmathew1992** [comment]
@robinbortlik  Oh, shoot!

I removed the unused method, this should be all set now.

**Comment by robinbortlik** [comment]
Hi @rmathew1992 . Thanks a lot for your hard work.

**Pull request merged**